0000-terraform-state-migration.md

```diff
@@ -0,0 +1,107 @@
+- Feature Name: `terraform_state_migration`
+- Start Date: 2019-07-03
+- RFC PR: [scipian/rfcs#0000](https://github.com/scipian/rfcs/pull/0000)
+- Community Issue: [scipian/community#6](https://github.com/scipian/community/issues/6)
+
+# Summary
+[summary]: #summary
+
+Migrating Terraform state from an on-boarding customer's AWS account to Scipian's
+managed backend is an important feature for users who are currently using Stack.
+This feature will allow them to seamlessly move over to the Scipian platform
+in a way that does not disrupt their current infrastructure.
+
+# Motivation
+[motivation]: #motivation
+
+There is currently no process for an on-boarding customer to migrate their
+Terraform state to Scipian. Having a smooth self-service process is important
+to encouraging users to use Scipian.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+After a new customer has been on-boarded, they will have the option to migrate
+their existing Terraform state to Scipian's backend. The process is simple:
+
+1. User will create 2 Kubernetes secrets, one with their AWS credentials for the 
+AWS acocunt they will be pulling their Terraform state from. The second secret
+they will create will contain temporary AWS credentials for Scipian's account,
+that will have minimal scope (only allowing a push to the Scipian backend 
+bucket), and will have a short TTL.
+2. User will then run a Kubernetes Job, based on a yaml template provided by
+the Scipian team. This job will run a special Docker image which will pull
+the Terraform state from the user's existing S3 bucket, and push it to Scipian's
+S3 backend.
+
+# Reference-level explanation
```

Comment:
What is the structure of the state stored in Scipian's S3 bucket? Mostly I'm wondering if the namespace is apart of the structure to avoid different teams accidentally overwriting each other's terraform state? Or maybe we will handle possible overwriting a different way?

Reply:
Would it be worth adding a mapping here from what the current stack state looks like to what the scipian state would look like?

Reply:
I didn't get into the specifics of Scipian's backend structure here, since this RFC will not change that. If we feel it would be helpful to specify that here, I'm happy to add it. It's `<scipian-bucket>/namespace/workspace/terraform.tfstate`